Whisky/Views/Bottle/Pins/PinCreationView.swift

```diff
@@ -0,0 +1,134 @@
+//
+//  PinCreationView.swift
+//  Whisky
+//
+//  This file is part of Whisky.
+//
+//  Whisky is free software: you can redistribute it and/or modify it under the terms
+//  of the GNU General Public License as published by the Free Software Foundation,
+//  either version 3 of the License, or (at your option) any later version.
+//
+//  Whisky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+//  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+//  See the GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License along with Whisky.
+//  If not, see https://www.gnu.org/licenses/.
+//
+
+import SwiftUI
+import UniformTypeIdentifiers
+import WhiskyKit
+
+struct PinCreationView: View {
+    var bottle: Bottle
+    @State var newPinName: String = ""
+    @State var newPinURL: URL = BottleData.defaultBottleDir
+    @State var pinPath: String = ""
+    @State var nameValid: Bool = false
+    @State var didError: Bool = false
+    @State var errorMessage: String = ""
+    @Environment(\.dismiss) var dismiss
+
+    var body: some View {
+        VStack {
+            HStack {
+                Text("pin.title")
+                    .bold()
+                Spacer()
+            }
+            Divider()
+            HStack(alignment: .top) {
+                Text("pin.name")
+                Spacer()
+                TextField(String(), text: $newPinName)
+                    .frame(width: 180)
+                    .textFieldStyle(.roundedBorder)
+                    .onChange(of: newPinName) { _, name in
+                        nameValid = !name.isEmpty
+                    }
+            }
+            HStack {
+                Text("pin.path")
+                Spacer()
+                Button("create.browse") {
+                    let panel = NSOpenPanel()
+                    panel.canChooseFiles = true
+                    panel.allowedContentTypes = [UTType.exe,
+                                                 UTType(exportedAs: "com.microsoft.msi-installer"),
+                                                 UTType(exportedAs: "com.microsoft.bat")]
+                    panel.directoryURL = bottle.url.appending(path: "drive_c")
+                    panel.canChooseDirectories = false
+                    panel.allowsMultipleSelection = false
+                    panel.canCreateDirectories = false
+                    panel.begin { result in
+                        if result == .OK {
+                            if let url = panel.urls.first {
+                                newPinURL = url
+                            }
+                        }
+                    }
+                }
+            }
+            HStack {
+                Text(pinPath)
+                    .truncationMode(.middle)
+                    .lineLimit(2)
+                    .help(pinPath)
```

Comment:
you can probably get rid of pinPath state and do a `let pinPath = newPinURL.prettyPath()` right here. Might help in the long-run maintainability of this feature.

Reply:
Had to do a bit of refactoring, but you're right, `pinPath` wasn't truly necessary. With some more refactoring, was able to remove a few redundant state variables in the latest commit.